File: backend/admin-ui/src/api/stats.ts
import client from './client'

export interface BackupInfo {
  filename: string
  size_kb: number
  mtime: string
}

export interface Stats {
  users_total: number
  users_active: number
  releases_total: number
  releases_active: number
  sessions_active: number
  recent_audit_count: number
  db_size_mb: number
  latest_backup: BackupInfo | null
}

export const statsApi = {
  async get(): Promise<Stats> {
    const r = await client.get('/admin/stats')
    return r.data
  },
}

File: backend/app/schemas/stats.py
"""stats schemas（Dashboard）"""
from datetime import datetime
from pydantic import BaseModel


class StatsResponse(BaseModel):
    users_total: int
    users_active: int            # is_active=True
    releases_total: int
    releases_active: int
    sessions_active: int         # revoked_at IS NULL
    recent_audit_count: int      # 最近 24h
    # 监控
    db_size_mb: float            # 数据库占用（MB）
    latest_backup: "BackupInfo | None"  # 最新备份状态


class BackupInfo(BaseModel):
    filename: str
    size_kb: float
    mtime: datetime


StatsResponse.model_rebuild()

